packages/core/src/Stores/common-store.js

```diff
@@ -8,6 +8,13 @@ import { clientNotifications } from './Helpers/client-notifications';
 export default class CommonStore extends BaseStore {
     constructor(root_store) {
         super({ root_store });
+
+        reaction(
+            () => this.app_routing_history.map(i => i.pathname),
```

Comment:
After reading the Mobx docs it seems your initial approach using `.length` would be more appropriate. The reason it didn't react to just `this.app_routing_history` is because the reference to the array would be the same, hence it would assume nothing was changed. They recommend observing a property instead, so `.length` is the way to go. Further reading:

- https://mobx.js.org/best/react.html#mobx-tracks-property-access-not-values

Reply:
The length of `app_routing_history` doesn't change as in `addRouteHistoryItem` function, the existing platform is spliced out and then unshifted again.